test/ui/form/form_view_model_test.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/model/survey_details_model.dart';
+import 'package:flutter_survey/ui/form/form_screen.dart';
+import 'package:flutter_survey/ui/form/form_state.dart';
+import 'package:flutter_survey/ui/form/form_view_model.dart';
+import 'package:flutter_survey/usecases/base/base_use_case.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../mocks/generate_mocks.mocks.dart';
+
+void main() {
+  group('FormViewModelTest', () {
```

Comment:
Also no test for `init` state, I think we can remove init state and start with loading state.